var/spack/repos/builtin/packages/py-pip/package.py

```diff
@@ -93,6 +93,15 @@ def install(self, spec, prefix):
         args = [os.path.join(whl, "pip")] + std_pip_args + ["--prefix=" + prefix, whl]
         python(*args)
 
+        # NOTE: Debian, and some other distros, inject `local` into the path in
+        # sysconfig, newer versions of pip (and pip's installer) apply this to the
+        # prefix path as well as the system path.  This works around it in a way that
+        # will let us largely ignore it.  See: https://github.com/spack/spack/pull/31939
+        local_path = os.path.join(prefix, "local")
+        if os.path.isdir(local_path) and not os.path.isdir(os.path.join(prefix, "lib")):
+            for p in os.listdir(local_path):
+                os.symlink(os.path.join(local_path, p), os.path.join(prefix, p))
```

Comment:
If you're going to be thorough, setuptools, wheel, and dozens of other packages that don't use `PythonPackage` need this as well:
```console
$ grep std_pip_args */package.py
cmor/package.py:            args = std_pip_args + ["--prefix=" + prefix, "."]
faiss/package.py:                args = std_pip_args + ["--prefix=" + prefix, "."]
fenics/package.py:                args = std_pip_args + ["--prefix=" + self.prefix, "."]
flatbuffers/package.py:                args = std_pip_args + ["--prefix=" + self.prefix, "."]
jsonnet/package.py:            args = std_pip_args + ["--prefix=" + self.prefix, "."]
mxnet/package.py:                args = std_pip_args + ["--prefix=" + prefix, "."]
py-dgl/package.py:            args = std_pip_args + ["--prefix=" + prefix, "."]
py-jaxlib/package.py:            args = std_pip_args + ["--prefix=" + self.prefix, "."]
py-keras/package.py:            args = std_pip_args + ["--prefix=" + prefix, "."]
py-or-tools/package.py:            args = std_pip_args + ["--prefix=" + prefix, "."]
py-pip/package.py:        args = [os.path.join(whl, "pip")] + std_pip_args + ["--prefix=" + prefix, whl]
py-setuptools/package.py:        args = ["-m", "pip"] + std_pip_args + ["--prefix=" + prefix, whl]
py-tensorboard-data-server/package.py:        args = std_pip_args + ["--prefix=" + prefix, wheel]
py-tensorflow-estimator/package.py:            args = std_pip_args + ["--prefix=" + prefix, "."]
py-tensorflow-hub/package.py:            args = std_pip_args + ["--prefix=" + prefix, "."]
py-tensorflow-probability/package.py:            args = std_pip_args + ["--prefix=" + prefix, "."]
py-tensorflow/package.py:            args = std_pip_args + ["--prefix=" + prefix, "."]
py-tfdlpack/package.py:            args = std_pip_args + ["--prefix=" + prefix, "."]
py-wheel/package.py:        args = std_pip_args + ["--prefix=" + prefix, self.stage.archive_file]
sgpp/package.py:            args = std_pip_args + ["--prefix=" + self.prefix, "."]
treelite/package.py:                args = std_pip_args + ["--prefix=" + self.prefix, "."]
```

Reply:
If these all have the same structure, should we make PythonManual (could use a better name) a new build system?

Reply:
Oof... some of these are much worse, Cmor is an autotools package that uses pip to install a module into its prefix, neither solution actually works for that without doing a view-like traversal.

Reply:
Curious what @alalazo thinks. We could extend both the current base class and PythonPackage and have this be a separate function that can be called in each package.

Reply:
Yeah, could do that.  Could make it part of an Executable wrapper: `pip_install` say so that the command includes it?  Even make it part of the existing callable `pip` from `setup_dependent_packages`?